hack/setup-monitoring.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# ----------------------------------------------------------------
+# OAuth2 Proxy Secrets
+# ----------------------------------------------------------------
+
+oauth2-secret() {
+  if [[ $# -ne 4 ]]; then
+    echo "invalid number of arguments"
+    echo "usage:"
+    echo "  $0 oauth2-secret NAME CLIENT_ID CLIENT_SECRET COOKIE_SECRET"
+    exit 1
+  fi
+
+  NAME=$1
+  CLIENT_ID=$2
+  CLIENT_SECRET=$3
+  COOKIE_SECRET=$4
+
+  oc create secret generic $NAME \
+    -n appstudio-workload-monitoring  \
+    --from-literal=client-id=$CLIENT_ID \
+    --from-literal=client-secret=$CLIENT_SECRET \
+    --from-literal=cookie-secret=$COOKIE_SECRET
+}
+
+# ----------------------------------------------------------------
+# Grafana Datasource Secrets
+# ----------------------------------------------------------------
+
+grafana-datasource-secret() {
+
+  if [[ $# -ne 3 ]]; then
+    echo "invalid number of arguments"
+    echo "usage:"
+    echo "  $0 grafana-datasource-secret NAME URL TOKEN"
+    exit 1
+  fi
+
+  NAME=$1
+  URL=$2
+  TOKEN=$3
+
+  DATA="apiVersion: 1
+datasources: 
+- name: $NAME
+  type: prometheus
+  access: proxy
+  url: https://$URL
+  basicAuth: false
+  withCredentials: true
+  isDefault: false
+  jsonData:
+    httpHeaderName1: 'Authorization'
+    timeInterval: 5s
+    tlsSkipVerify: true
+  secureJsonData:
+    httpHeaderValue1: 'Bearer $TOKEN'"
+  
+  oc create secret generic $NAME -n appstudio-workload-monitoring --from-literal=$NAME.yaml="$DATA" 
```

Comment:
the same as for `create` vs `apply`

Reply:
keeping the `oc create secret` command as it suppors the `--from-literal` flags, but using `--dry-run=client -o yaml` options to generate the YAML representation and `oc apply -f -` to apply if afterwards.
see c3454ee